pkg/controller/operandrequest/reconcile_operator.go

```diff
@@ -161,8 +159,38 @@ func (r *ReconcileOperandRequest) updateSubscription(cr *operatorv1alpha1.Operan
 	return nil
 }
 
+func preDeleteCheck(operandName string, reconcileReq reconcile.Request, registryInstance *operatorv1alpha1.OperandRegistry) (int, int) {
+	klog.V(4).Infof("Pre-check for delete subscription: $s", operandName)
+	operatorStatus := registryInstance.Status.OperatorsStatus[operandName]
+	if operatorStatus.Phase != operatorv1alpha1.OperatorReady {
+		pos := registryInstance.GetReconcileRequest(operandName, reconcileReq)
+		reconcileReqNum := len(operatorStatus.ReconcileRequests)
+		return pos, reconcileReqNum
+	}
+	return -1, 0
+}
+
 func (r *ReconcileOperandRequest) deleteSubscription(operandName string, requestInstance *operatorv1alpha1.OperandRequest, registryInstance *operatorv1alpha1.OperandRegistry, configInstance *operatorv1alpha1.OperandConfig, reconcileReq reconcile.Request) error {
 	klog.V(2).Info("Delete Subscription: ", operandName)
+	pos, reconcileReqNum := preDeleteCheck(operandName, reconcileReq, registryInstance)
+
+	// If subscription not find in the registry status, nothing to do, return nil
+	if pos == -1 {
+		return nil
+	}
+	// If there are more than one reconcile request in the registry status, just update the
+	// operandrequest and operandregistry status, don't really delete the subscription.
+	if reconcileReqNum > 1 {
+		requestInstance.CleanMemberStatus(operandName)
+		if err := r.client.Status().Update(context.TODO(), requestInstance); err != nil {
+			return err
+		}
+		if err := r.deleteRegistryStatus(registryInstance, reconcileReq, operandName); err != nil {
```

Comment:
Why delete registry status here?

Reply:
For example, iam operator requested by 2 `OperandRequest`, `ibm-common-services/common-service` and `ibm-common-services/test-req-2`, if we delete iam operator from request `ibm-common-services/test-req-2`, but actually, iam operator still used by request `ibm-common-services/common-service`, so we don't really delete this operator, just remove the status from registry cr.  When all the `reconcileRequests` deleted, this operator will be real delete.
Before delete iam operator from request `ibm-common-services/test-req-2`
```yaml
  status:
    operatorsStatus:
      ibm-iam-operator:
        phase: Running
        reconcileRequests:
        - name: common-service
          namespace: ibm-common-services
        - name: test-req-2
          namespace: ibm-common-services
```

After delete operator iam
```yaml
  status:
    operatorsStatus:
      ibm-iam-operator:
        phase: Running
        reconcileRequests:
        - name: common-service
          namespace: ibm-common-services
```

Reply:
Thanks @DanielXLee got it. 

Can you please put this in your code as comments?

Reply:
ok